File: http-server-function/src/test/java/io/micronaut/http/server/util/TestController.java
package io.micronaut.http.server.util;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;

@Controller
public class TestController {
    @Get
    @Produces(MediaType.TEXT_PLAIN)
    public HttpResponse<?> simpleGet() {
        return HttpResponse.ok().body("foo");
    }

    @Get("/no-type")
    public HttpResponse<?> simpleGetNoContentType() {
        return HttpResponse.ok().body("foo");
    }

    @Get("/json")
    public HttpResponse<?> simpleJsonGet() {
        return HttpResponse.ok().body(new SimpleModel("foo"));
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/server/response/TestController.kt
package io.micronaut.docs.server.response

import io.micronaut.http.HttpResponse
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces

//tag::clazz[]
@Controller("/test")
class TestController {

    @Get
    fun index(): HttpResponse<*> {
        return HttpResponse.ok<Any>().body("{\"msg\":\"This is JSON\"}")
    }

    @Produces(MediaType.TEXT_HTML) // <1>
    @Get("/html")
    fun html(): String {
        return "<html><title><h1>HTML</h1></title><body></body></html>"
    }
}
//end::clazz[]